src/main/java/mate/academy/spring/security/EmailValidator.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.security;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import mate.academy.spring.security.annotations.ValidEmail;
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
+    private Pattern pattern;
+    private Matcher matcher;
+    private static final String EMAIL_PATTERN = "^[_A-Za-z0-9-+]+(.[_A-Za-z0-9-]+)*@"
+            + "[A-Za-z0-9-]+(.[A-Za-z0-9]+)*(.[A-Za-z]{2,})$";
+
+    @Override
+    public void initialize(ValidEmail constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext context) {
```

Comment:
@TatianaPirch This is optional. Can you cover this with JUnit tests?

Reply:
Today, my favorite topic is multithreading :)
I will try to do it during the weekend. Ok?

Reply:
np)